OrbitGl/TimerTrack.cpp

```diff
@@ -0,0 +1,302 @@
+// Copyright (c) 2020 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "TimerTrack.h"
+
+#include <limits>
+
+#include "Capture.h"
+#include "EventTrack.h"
+#include "FunctionUtils.h"
+#include "GlCanvas.h"
+#include "TextBox.h"
+#include "TimeGraph.h"
+#include "absl/flags/flag.h"
+#include "absl/strings/str_format.h"
+
+// TODO: Remove this flag once we have a way to toggle the display return values
+ABSL_FLAG(bool, show_return_values, false, "Show return values on time slices");
+
+//-----------------------------------------------------------------------------
+TimerTrack::TimerTrack(TimeGraph* time_graph) : Track(time_graph) {
+  text_renderer_ = time_graph->GetTextRenderer();
+
+  num_timers_ = 0;
+  min_time_ = std::numeric_limits<TickType>::max();
+  max_time_ = std::numeric_limits<TickType>::min();
+}
+
+//-----------------------------------------------------------------------------
+void TimerTrack::Draw(GlCanvas* canvas, PickingMode picking_mode) {
+  float track_height = GetHeight();
+  float track_width = canvas->GetWorldWidth();
+
+  SetPos(canvas->GetWorldTopLeftX(), m_Pos[1]);
```

Comment:
I know you didn't introduce this, but I find this pattern to update pos and size in the draw() method not ideal. Both should not be influenced by the actual drawing, but rather updated when the underlying data or layouting change that forces this occurs. This may also happen when the view is scrolled to adjust the track height only to the actual visible timings if this is the desired behavior.

With the current pattern, I cannot get positioning information about the track outside of the draw() method, or only if I am certain that drawing has happened at least once after the last layout-influencing changes, and I have no way to force this since I can't just trigger a draw() at random. In general, the draw() method comes with side effects I would not expect as a user of this class.

I am not sure if this is something we can or even want to address, and if so probably not in this PR, but for me, these kind of side-effects made it hard to understand what's happening when I first started reading the Orbit code.

Reply:
Good point, let's follow up on this. I think it's out of scope for this PR (and I wouldn't even know how to address this without looking into it, maybe it's easy, maybe it's hard).